core-bundle/src/EventListener/CsrfTokenCookieSubscriber.php

```diff
@@ -102,21 +102,36 @@ public static function getSubscribedEvents(): array
 
     private function requiresCsrf(Request $request, Response $response): bool
     {
-        foreach ($request->cookies as $key => $value) {
-            if (!$this->isCsrfCookie($key, $value)) {
+        $requestCookies = $request->cookies->all();
+        $responseCookies = $response->headers->getCookies(ResponseHeaderBag::COOKIES_FLAT);
+
+        // Ignore any cookies in the request and response that are being deleted (#7344)
+        $responseCookies = array_filter(
+            $responseCookies,
+            static function (Cookie $responseCookie) use ($requestCookies): bool {
```

Comment:
```suggestion
            static function (Cookie $responseCookie) use (&$requestCookies): bool {
```
Otherwise `unset($requestCookies[…]);` has no effect outside the filter.

Reply:
Hm, true, but weird as it worked in my test 🤔. Need to check if there was a flaw in my test.